datafusion/substrait/tests/testdata/test_plans/intersect.substrait.json

```diff
@@ -0,0 +1,118 @@
+                      ]
+                    }
+                  },
+                  "input": {
+                    "read": {
+                      "common": {
+                        "direct": {}
+                      },
+                      "baseSchema": {
+                        "names": [
+                          "a"
+                        ],
+                        "struct": {
+                          "types": [
+                            {
+                              "i64": {
+                                "nullability": "NULLABILITY_NULLABLE"
+                              }
+                            }
+                          ],
+                          "nullability": "NULLABILITY_NULLABLE"
+                        }
+                      },
+                      "namedTable": {
+                        "names": [
+                          "data"
+                        ]
+                      }
+                    }
+                  },
+                  "expressions": [
+                    {
+                      "selection": {
+                        "directReference": {
+                          "structField": {}
+                        },
+                        "rootReference": {}
+                      }
+                    }
+                  ]
+                }
+              },
+              {
+                "project": {
+                  "common": {
+                    "emit": {
+                      "outputMapping": [
+                        1
+                      ]
+                    }
+                  },
+                  "input": {
+                    "read": {
+                      "common": {
+                        "direct": {}
+                      },
+                      "baseSchema": {
+                        "names": [
+                          "a"
+                        ],
+                        "struct": {
+                          "types": [
+                            {
+                              "i64": {
+                                "nullability": "NULLABILITY_NULLABLE"
+                              }
+                            }
+                          ],
+                          "nullability": "NULLABILITY_NULLABLE"
+                        }
+                      },
+                      "namedTable": {
+                        "names": [
+                          "data2"
+                        ]
+                      }
+                    }
+                  },
+                  "expressions": [
+                    {
+                      "selection": {
+                        "directReference": {
+                          "structField": {}
+                        },
+                        "rootReference": {}
+                      }
+                    }
+                  ]
+                }
+              }
+            ],
+            "op": "SET_OP_INTERSECTION_PRIMARY"
```

Comment:
If the intent is for this plan to correspond to
```sql
SELECT a FROM data INTERSECT SELECT a FROM data2"
```
this should actually be `SET_OP_INTERSECTION_MULTISET`. See relevant [docs].(https://substrait.io/relations/logical_relations/#set-operation-types)

This is tied to a relatively recent spec change: https://github.com/substrait-io/substrait/pull/708

Reply:
no, it's the other way around. the intent is to add consumer rule for `SET_OP_INTERSECTION_PRIMARY` which isn't generally equivalent to `INTERSECT` op, but it is when used with 2 inputs, which is what's being tested here.

`SET_OP_INTERSECTION_MULTISET` should also be handled of course, but that would be in addition to what's being added here, not instead of, right?

Reply:
> the intent is to add consumer rule for SET_OP_INTERSECTION_PRIMARY which isn't generally equivalent to INTERSECT op, but it is when used with 2 inputs,

Gotcha, that is correct for that specific case. It ends up being a little confusing because in the associated test
```
    assert_substrait_sql(
        proto_plan,
        "SELECT a FROM data INTERSECT SELECT a FROM data2",
    )
    .await
```
The substrait plan ends up being _semantically_ equivalent to the SQL query, but is not the correct encoding generally, but I don't think that's a blocker.

Reply:
thanks, I'll add other ops in follow-ups (primary with multiple rels and multisets) and that should clear the confusion, I think. The sql query used to test "primary with multiple rels" will probably be some combination of intersect and unions, right?

Reply:
> The sql query used to test "primary with multiple rels" will probably be some combination of intersect and unions, right?

I believe that SET_OP_INTERSECTION_PRIMARY with primary input `<A>` and secondary inputs `<B>`, `<C>`, ... ends up being equivalent to something like:
```sql
<A> INTERSECT DISTINCT (<B> UNION <C> UNION ...)
```
which satisfies the requirement that it outputs a row from `<A>` if is is present in _any_ of the secondary inputs.